graphql-async/test/async_test.ml

```diff
@@ -16,14 +16,26 @@ let test_query schema ctx query expected =
     match Graphql_parser.parse query with
     | Error err -> failwith err
     | Ok doc ->
-      Graphql_async.Schema.execute schema ctx doc >>| fun result ->
-      let result' = match result with
-      | Ok (`Response data) -> data
-      | Error err -> err
-      in
-      Alcotest.check yojson "invalid execution result" expected result'
+      Graphql_async.Schema.execute schema ctx doc >>= (function
+      | Ok (`Response data) -> Async_kernel.return data
+      | Ok (`Stream stream) ->
+          Async_kernel.Pipe.to_list stream >>| fun lst ->
+            `List (
+              List.fold_right (fun x acc -> match x with
+                | Ok data -> data :: acc
+                | _ -> acc) lst [])
+      | Error err -> Async_kernel.return err)
+      >>| fun result ->
+        Alcotest.check yojson "invalid execution result" expected result
   end
 
+let take n l =
+  let rec loop n l = match n, l with
+    | 0, _ | _, [] -> []
+    | _, x::xs -> x :: loop (n - 1) xs
+  in
+  loop n l
+
```

Comment:
Is it worthwhile to include this compared to just returning the full list from the `int_stream` field?